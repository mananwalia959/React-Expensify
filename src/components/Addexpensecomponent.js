import React from 'react';
import ExpenseForm from './ExpenseForm';
import {connect} from 'react-redux';
import {startaddExpense} from '../actions/expenses';
export class Addexpensecomponent extends React.Component {
    onSubmit = (expense) => {
        this.props.startaddExpense(expense);
        this.props.history.push('/');

    }
    render() {
        return (
            <div classname='page-header'>
            <div className='content-container'>
            <h1 className='page-header__title'>Add Transaction</h1>
            </div>
            <div className='content-container'>
            <ExpenseForm 
            onSubmit={this.onSubmit}
            />
            </div>
    </div>

        )
    }
    
}

const mapDispatchToProps=(dispatch) => ({
    startaddExpense:(expense) => dispatch(startaddExpense(expense))
})


export default connect(undefined,mapDispatchToProps)(Addexpensecomponent);