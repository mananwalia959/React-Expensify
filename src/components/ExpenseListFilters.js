import React from 'react';
import {connect} from 'react-redux';
import {setTextFilter, sortByAmount, sortByDate,setstartDate,setendDate,settransactiontype} from '../actions/filters'
import {DateRangePicker} from 'react-dates'

export class ExpenseListFilters extends React.Component{
    state={
        calendarFocused:null
    }
    onDatesChange=({startDate,endDate}) => {
        this.props.setstartDate(startDate);
        this.props.setendDate(endDate);
    };
    onFocusChange = (calendarFocused)=> {
        this.setState(()=>({calendarFocused}) )
    }
    onTextChange=(e) =>{
        this.props.setTextFilter(e.target.value)
    }
    onSortChange=(e) =>{
        if(e.target.value==='date'){
        this.props.sortByDate();
        }
        else if(e.target.value==='amount'){
        this.props.sortByAmount();
        }
    }
    onTransactionTypeChange=(e) =>{
        const option=e.target.value
        this.props.settransactiontype(option)
    }


    render(){
        return(
        <div className='content-container'>
        <div className='input-group'>
       
        <div className='input-group__item'>
            <select 
            className='select--secondary'
            value={this.props.filters.sortby} 
            onChange={this.onSortChange}>
            <option value = 'date' > Date </option>
            <option value = 'amount'> Amount </option>
            </select>
        </div>
        <div className='input-group__item'>
            <select
            className= 'select--secondary' 
            value={this.props.filters.transactiontype} 
            onChange={this.onTransactionTypeChange}>
            <option value = 'both' > Both </option>
            <option value = 'expense'> Expense </option>
            <option value = 'income' > Income </option>
            </select>
        </div>
        </div>
        <div className="input-group">
        <div className='input-group__item'>
        <input 
        type ='text'
        className='text-input' 
        placeholder='Search transactions'
        value={this.props.filters.text} 
        onChange={this.onTextChange} 
        />
        </div>

        <div className='input-group__item'>
        <DateRangePicker
        startDate={this.props.filters.startDate}
        startDateId={this.props.filters.startDate?this.props.filters.startDate.toString():'No_Start'}
        endDate={this.props.filters.endDate}
        endDateId={this.props.filters.endDate?this.props.filters.endDate.toString():"No_END"}
        onDatesChange={this.onDatesChange}
        focusedInput={this.state.calendarFocused}
        onFocusChange={this.onFocusChange}
        showClearDates={true}
        numberOfMonths={1}
        isOutsideRange={()=>false}
        />
        </div>
        </div>
        </div>
        )
  
    }
}
const MapStatetoProps=(state) => ({
    filters:state.filters
});

const MapDispatchToProps=(dispatch) => ({
    setTextFilter: (text)=> dispatch(setTextFilter(text)),
    sortByDate: () => dispatch(sortByDate()),
    sortByAmount:() => dispatch(sortByAmount()),
    setstartDate:(startDate)=>dispatch(setstartDate(startDate)),
    setendDate:(endDate)=> dispatch(setendDate(endDate)),
    settransactiontype:(tranaction)=> dispatch(settransactiontype(tranaction))
})


export default connect (MapStatetoProps,MapDispatchToProps)(ExpenseListFilters);
