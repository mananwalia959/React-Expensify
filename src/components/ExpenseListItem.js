import React from 'react';
import {Link} from 'react-router-dom'
import moment from 'moment'



 const ExpenseListItem =({id,description,amount,createdAt,transaction}) => (
    <Link className='list-item' to={`/edit/${id}`}>
    <div>
    <h3 className='list-item__title'>{transaction[0].toUpperCase()+ transaction.slice(1)}:{description}</h3>
    <span className='list-item__sub-title'>{moment(createdAt).format('MMMM Do, YYYY')}</span>
    </div>
    <div>
    <h3 className='list-item__data'> Rs.{(amount/100)}</h3>
    </div>
    </Link>
)

export default ExpenseListItem;


//    <Link to={'/edit/'+id}>