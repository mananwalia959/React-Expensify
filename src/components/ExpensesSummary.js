import React from 'react';
import {connect} from 'react-redux';
import {Link} from 'react-router-dom'
import selectExpenses from '../selectors/expenses';
import selectExpensesTotal from '../selectors/expensestotal';

export const ExpensesSummary = ({expenseCount,transactiontotals}) => {
    const expenseWord = expenseCount === 1 ? "transaction":"transactions";
    const formattedExpenseTotal = (transactiontotals.total_expense/100)
    const formattedIncomeTotal   = (transactiontotals.total_income/100)


    return (
        <div className='page-header'>
         <div className='content-container'>
          <h1 className='page-header__title'> Viewing <span>{expenseCount}</span> {expenseWord} with Net Amount = <span>{formattedIncomeTotal-formattedExpenseTotal}</span> <br/>Expenses =<span>Rs.{formattedExpenseTotal}</span> and Incomes = <span>Rs.{formattedIncomeTotal}</span> </h1>
           <div>
            <Link className ='button' to='/create'>Add Transaction</Link>
           </div>
         </div>
        </div>
    )
}

const mapStateToProps = (state) => {
    const visibleExpenses = selectExpenses(state.expenses, state.filters);
  
    return {
      expenseCount: visibleExpenses.length,
      transactiontotals: selectExpensesTotal(visibleExpenses)
    };
  };
  
  export default connect(mapStateToProps)(ExpensesSummary);