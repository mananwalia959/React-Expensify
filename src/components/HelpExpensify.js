import React from 'react';

const HelpExpensify =() => (
    <div>
        Help page
    </div>
)

export default HelpExpensify;