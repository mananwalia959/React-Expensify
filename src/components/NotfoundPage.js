import React from 'react';
import {Link} from 'react-router-dom';

const NotfoundPage =() => (
    <div>
        404 : <Link to='/'>Homepage</Link>
    </div>
)

export default NotfoundPage;
