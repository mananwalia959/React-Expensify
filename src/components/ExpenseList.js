import React from 'react';
import {connect} from 'react-redux';
import ExpenseListItem from './ExpenseListItem';
import SelectExpenses from '../selectors/expenses';
export const ExpenseList = (props) => (
    <div className='content-container'>
      <div>
       <div className='list-header'>
        <div className='show-for-mobile'>Expenses</div>
        <div className='show-for-desktop'>Expense</div>
        <div className='show-for-desktop'>Amount</div>    
      </div>
    </div>


    <div className='list-body'>
    {
        props.expenses.length === 0?(
            <div className='list-item listitem--message'>
            <span>No Transactions</span>
            </div>
        ):(
            props.expenses.map((expense) => {
                return <ExpenseListItem key = {expense.id} {...expense}  />
            })
        )
    }
    </div> 
   
    </div>
)

const MapStateToProps = (state)=>{
    return {
        expenses:SelectExpenses(state.expenses,state.filters)
    }
}
export default connect(MapStateToProps)(ExpenseList);


