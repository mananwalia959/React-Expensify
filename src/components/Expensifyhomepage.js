import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters'
const Expensifyhomepage = () => (
    <div>
        <ExpenseListFilters />   
        <ExpenseList />
    </div>
);


export default Expensifyhomepage;