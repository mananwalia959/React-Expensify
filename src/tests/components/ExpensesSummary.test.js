import React from 'react';
import {shallow} from 'enzyme';
import {ExpensesSummary} from '../../components/ExpensesSummary';

test('should correctly render ExpensesSummary with 1 expense', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={1} transactiontotals={{
        total_expense:456,
        total_income:123
    }} />);
    expect(wrapper).toMatchSnapshot();
  });
  
  test('should correctly render ExpensesSummary with multiple expenses', () => {
    const wrapper = shallow(<ExpensesSummary expenseCount={23} transactiontotals={{
        total_expense:123,
        total_income:456
    }} />);
    expect(wrapper).toMatchSnapshot();
  });