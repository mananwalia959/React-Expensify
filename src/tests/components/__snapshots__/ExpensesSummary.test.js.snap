// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render ExpensesSummary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
       Viewing 
      <span>
        1
      </span>
       
      transaction
       with Net Amount = 
      <span>
        -3.3299999999999996
      </span>
       
      <br />
      Expenses =
      <span>
        Rs.
        4.56
      </span>
       and Incomes = 
      <span>
        Rs.
        1.23
      </span>
       
    </h1>
    <div>
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Transaction
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render ExpensesSummary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
       Viewing 
      <span>
        23
      </span>
       
      transactions
       with Net Amount = 
      <span>
        3.3299999999999996
      </span>
       
      <br />
      Expenses =
      <span>
        Rs.
        1.23
      </span>
       and Incomes = 
      <span>
        Rs.
        4.56
      </span>
       
    </h1>
    <div>
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Transaction
      </Link>
    </div>
  </div>
</div>
`;
